Modules/_ssl.c

```diff
@@ -3281,6 +3295,73 @@ set_verify_flags(PySSLContext *self, PyObject *arg, void *c)
     return 0;
 }
 
+/* Getter and setter for protocol version */
+#if defined(SSL_CTRL_GET_MAX_PROTO_VERSION)
+
+static int
+set_min_max_proto_version(PySSLContext *self, PyObject *arg, int what)
+{
+    long v;
+    int result;
+
+    if (!PyArg_Parse(arg, "l", &v))
+        return -1;
+    if (v > INT_MAX) {
+        PyErr_SetString(PyExc_OverflowError, "Option is too long");
+        return -1;
+    }
+    if ((v == PY_PROTO_MINIMUM_SUPPORTED) ||
+            (v == PY_PROTO_MAXIMUM_SUPPORTED)) {
+        /* 0: set lowest or highest supported version */
```

Comment:
This is confusing to me, how does it know the difference between `set_min(min)` and `set_min(max)`?

Reply:
``SSL_CTX_set_min_proto_version(ctx, 0)`` sets lowest version, ``SSL_CTX_set_max_proto_version(ctx, 0)`` sets highest version. The function selects betwen min and max based on the ``what`` argument.

(Q.E.D. why I prefer highest/lowest over max/min) :p

```
    if (what == 0) {
        result = SSL_CTX_set_min_proto_version(self->ctx, v);
    }
    else {
        result = SSL_CTX_set_max_proto_version(self->ctx, v);
    }
```

Reply:
How do you set the minimum version to be the same as the maximum version?